simulator.py

```diff
@@ -0,0 +1,172 @@
+"""
+Create a simulator to calculate pnl
+"""
+
+import collections
+import copy
+import uuid
+from thousandaire.constants import TRADING_CONFIGS
+from thousandaire.dataset import Data
+
+DATASET = 'price_dataset'
+COST_FUNC = 'cost_function'
+
+class SimulateMethod():
+    """
+    The method for simulating.
+    """
+    def __init__(self, region, instruments):
+        self.quantity = collections.namedtuple('value', ['value', 'quantity'])
+        self.cur_asset = {
+            instrument : self.quantity(0, 0) for instrument in instruments}
+        self.cur_asset[region] = self.quantity(1, 1)
+        self.__portfolio = list()
+        self.__pnl = Data('pnl', ['pnl', 'trading_cost'])
```

Comment:
These are all results of simulation. Can we put them into a dict named `results`? (`self.__results`)